File: program_data/github_java_sort_function/6/39.java
@SuppressWarnings({ "rawtypes", "unchecked" })
        
        public void radixsorter(int[] a) throws FileNotFoundException
        {
                final int RADIX = 10;
         
        
                List[] bucket = new ArrayList[RADIX];
      
                for(int i=0; i < RADIX; i++)
                {
                    bucket[i] = new ArrayList();
                }
        
                boolean maxLength = false;
                int tmp = 0;
                int place = 1;
        
                while(!maxLength)
                {
                    maxLength = true;
           
                    for(Integer i : a)
                    {
                        tmp = i / place;
                        bucket[tmp % RADIX].add(i);
                
                        if(maxLength && tmp > 0)
                        {
                        maxLength = false;
                        }
                    }
            
                    
                    int idx = 0;
                    
                    for(int b = 0; b < RADIX; b++)
                    {
                        for(int j = 0; j < bucket[b].size(); j++)
                        {
                            a[idx++] = (int) bucket[b].get(j);
                        }
                        
                        bucket[b].clear();
                    }
                    
                    place *= RADIX;
               }
       }
}

File: program_data/PKU_raw/37/39.c
char s[100001];
int bucket[26];
int t, i, j, k, len;
int main(void)
{
	scanf("%d", &t);
	for(i=0;i<t;i++)
	{
		for(k=0;k<26;k++)bucket[k]=0;
		scanf("%s", s);
		while(s[j]!='\0')
		{
			bucket[s[j]-'a']++;
			j++;
		}
		len=j;
		j=0;
		while(s[j]!='\0')
			if(bucket[s[j]-'a']==1)
			{
				printf("%c\n", s[j]);
				break;
			}
			else j++;
		if(j==len)printf("no\n");
		j=0;
	}
	return 0;
}

